tensorflow/core/kernels/data/experimental/random_dataset_op_test.cc

```diff
@@ -0,0 +1,237 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/core/kernels/data/experimental/random_dataset_op.h"
+
+#include "tensorflow/core/kernels/data/dataset_test_base.h"
+#include "tensorflow/core/lib/random/philox_random.h"
+#include "tensorflow/core/lib/random/random_distributions.h"
+
+namespace tensorflow {
+namespace data {
+namespace experimental {
+namespace {
+
+constexpr char kNodeName[] = "random_dataset";
+constexpr char kIteratorPrefix[] = "Iterator";
+
+// Number of random samples generated per test
+constexpr int kCount = 10;
+
+// Generate the first `count` random numbers that the kernel should produce
+// for a given seed/seed2 combo.
+// For compatibility with the test harness, return value is a vector of scalar
+// Tensors.
+std::vector<Tensor> GenerateExpectedData(int64 seed, int64 seed2, int count) {
+  std::vector<Tensor> ret;
+  auto parent_generator = random::PhiloxRandom(seed, seed2);
+  auto generator =
+      random::SingleSampleAdapter<random::PhiloxRandom>(&parent_generator);
+
+  for (int i = 0; i < count; ++i) {
+    ret.push_back(CreateTensor<int64>(TensorShape({}), {generator()}));
+  }
+  return ret;
+}
+
+class RandomDatasetParams : public DatasetParams {
+ public:
+  RandomDatasetParams(int64 seed, int64 seed2, DataTypeVector output_dtypes,
+                      std::vector<PartialTensorShape> output_shapes,
+                      string node_name)
+      : DatasetParams(std::move(output_dtypes), std::move(output_shapes),
+                      std::move(node_name)),
+        seed_(CreateTensor<int64>(TensorShape({}), {seed})),
+        seed2_(CreateTensor<int64>(TensorShape({}), {seed2})) {}
+
+  virtual Status GetInputPlaceholder(
+      std::vector<string>* input_placeholder) const override {
+    *input_placeholder = {RandomDatasetOp::kSeed, RandomDatasetOp::kSeed2};
+    return Status::OK();
+  }
+
+  virtual Status GetInputs(
```

Comment:
@frreiss After merging #32750 , `GetInputs()` is changed to `std::vector<Tensor> GetInputTensors() const override`, which only returns the input tensors (no need to return the dataset tensor). [Here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/data/experimental/sampling_dataset_op_test.cc#L40-L45) is an example. For this case, it will be:
```C++
std::vector<Tensor> GetInputTensors() const override {
  return {seed_, seed2_};
}

Reply:
Fixed this in commit bb2a02e